src/components/UserGroupAccess/Autocomplete/AccessAutocomplete.js

```diff
@@ -0,0 +1,79 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import PropTypes from 'prop-types'
+import React, { useState, useEffect, useCallback } from 'react'
+import { debounce } from '../helper'
+import { Autocomplete } from './Autocomplete'
+
+const query = {
+    search: {
+        resource: 'userGroups',
+        params: ({ search }) => ({
+            query: search,
+        }),
+    },
+}
+
+export const AccessAutocomplete = ({ selected, onSelection }) => {
+    const [search, setSearch] = useState('')
+    const [showResults, setShowResults] = useState(false)
+    const { data, refetch, fetching } = useDataQuery(query, {
+        lazy: true,
+        onComplete: () => setShowResults(true),
+    })
+
+    /**
+     * On selection this syncs the displayName of the parent selection to the local input state.
+     */
+
+    useEffect(() => setSearch(selected), [selected])
+
+    /**
+     * If the search string changes and is truthy, send out a request, otherwise
+     * clear the selection.
+     */
+
+    const debouncedRefetch = useCallback(debounce(refetch, 250), [refetch])
+
+    useEffect(() => {
+        if (search && search === selected) {
+            return
+        }
+
+        if (search) {
+            debouncedRefetch({ search })
+        } else {
+            onSelection(null)
+            setShowResults(false)
+        }
+    }, [search])
+
+    // Concatenate all the results
+    let results = []
+
+    if (data?.search?.userGroups) {
+        results = data?.search?.userGroups
```

Comment:
Currently, it is possible to select the same group of users multiple times. Should the option be filtered out of the `searchResults` if it has already been added? Additionally, the `Remove access` button behaves strangely in this case and removes all the occurrences in the list. Can you have a look?

https://github.com/dhis2/apk-distribution-app/assets/6817073/12b81875-e42d-4c6a-ada2-4264d87a3eae

Reply:
By filtering out of the `searchResults` the user groups that have been already added the strange behavior of  `Remove access` is fixed

Reply:
Filtering options already used
![image](https://github.com/dhis2/apk-distribution-app/assets/29384664/e02dbdbb-f18a-463b-b164-70703f1e278b)